app/lib/bing_api/client_v7.rb

```diff
@@ -0,0 +1,102 @@
+module BingApi
+  class ClientV7
+    include HTTParty
+    base_uri 'https://api.cognitive.microsoft.com'
+    default_timeout 1 # hard timeout after 1 second
+
+    attr_reader :options
+
+    class << self
+      def base_path
+        "/bing/v7.0/search"
+      end
+    end
+
+    def api_key
+      Rails.application.secrets.bing_api_v7
+    end
+
+    def api_key_hash
+      { "Ocp-Apim-Subscription-Key": api_key }
+    end
+
+    def pages_process_key1
+      'webPages'
+    end
+
+    def pages_process_key2
+      'value'
+    end
+
+    def initialize(options)
+      @options = options
+    end
+
+    def search
+      handle_timeouts do
+        handle_caching do
+          self.class.get(build_url_from_options, headers: api_key_hash, format: :json)
+        end
+      end
+    end
+
+    def pages_process
+      data = search
+      return unless pages_data?(data)
+      data[pages_process_key1][pages_process_key2].each do |page_data|
+        company.pages.create!(
+          title: page_data['name'],
+          source_url: page_data['url'],
+          page_type: Page::BING_TYPE,
+          status: Page::PENDING_STATUS
+        )
+      end
+    end
+
+    private
+
+    def handle_timeouts
+      yield
+    rescue Net::OpenTimeout, Net::ReadTimeout => e
+      Rails.logger.warn("Handle_timeouts #{e}")
+      {}
+    end
+
+    def cache_key
+      if options[:company_id]
+        "BingApi:ClientV7:company_id:#{ options[:company_id] }"
+      # elsif
+      # add extra options[:parameters] when exist
+      end
+    end
+
+    def handle_caching
+      redis = Redis.new
+      if cached = redis.get(cache_key)
+        JSON[cached]
+      else
+        yield.tap do |results|
+          redis.setex(cache_key, 3600, results.to_json)
```

Comment:
do you really need this garbage in a cache?
it's only one time data load.. 
btw, there's no need to cache this data, as for each future processing it'll require the most recent data

Reply:
actually caching was not a requirement and could be deleted.
rule is simple - less code = less bugs.

PS: @romaslmd my idea is just to move this code as soon as possible to master. It's already took too much time.